### PR TITLE
GET All Test Cases XAPTC-43

### PR DESCRIPTION
**Pull request by frankquinn**

### CODE REVIEW

File: uiServices/src/testCaseServices.go [hunk 1]
```diff
@@ -113,4 +127,63 @@ func putTestCase(rw http.ResponseWriter, req *http.Request) {
...
+	testCasePathDir := getTestCaseHeader(req)
+	if len(testCasePathDir) <= 1 {
+		logrus.Error("No file directory entered")
+		rw.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	files, err := ioutil.ReadDir(testCasePathDir)
+	if err != nil {
+		log.Fatal(err)
```
Comment (ConorSmyth) [inline uiServices/src/testCaseServices.go]: We should not use fatal here

File: uiServices/src/testCaseServices.go [hunk 2]
```diff
@@ -113,4 +127,63 @@ func putTestCase(rw http.ResponseWriter, req *http.Request) {
...
+	for _, file := range files {
+		if filepath.Ext(testCasePathDir +file.Name()) == ".xml" {
+
+			testCase := new(testStrategies.TestDefinition)
+
+			filename := file.Name()
+
+			file, err := os.Open(fmt.Sprintf("%s%s", testCasePathDir, filename))
+			if err != nil {
+				continue
```
Comment (ConorSmyth) [inline uiServices/src/testCaseServices.go]: I think that we should have log stmts here

Comment (ConorSmyth) [reply uiServices/src/testCaseServices.go]: Also if there is an error we are not setting the response header

Comment (frankquinn) [reply uiServices/src/testCaseServices.go]: I'll get to this soon. I'll add in log statements but jenny told me to leave out the response headers for each.

**Comment by frankquinn** [comment]
Added Log statements and now returning multiple objects in Get All Test

File: uiServices/src/testCaseServices.go [hunk 3]
```diff
@@ -113,4 +126,66 @@ func putTestCase(rw http.ResponseWriter, req *http.Request) {
...
+	for _, file := range files {
+		if filepath.Ext(testCasePathDir +file.Name()) == ".xml" {
+
+			testCase := new(testStrategies.TestDefinition)
+
+			filename := file.Name()
+
+			file, err := os.Open(fmt.Sprintf("%s%s", testCasePathDir, filename))
+			if err != nil {
+				logrus.Error("Cannot Open File: "+ filename)
```
Comment (ConorSmyth) [inline uiServices/src/testCaseServices.go]: If the any of these conditions are true I would not expect the flow to continue, I would expect a log message and setResponse and error. If you can't read or unmarshall it is unrecoverable. 

```
	logrus.Error("Cannot Unmarshall")
		rw.WriteHeader(http.StatusBadRequest)
		return
```

Comment (AlexeiSuleac) [reply uiServices/src/testCaseServices.go]: What happens if one of the files is being used by another user and cannot be opened? But if you need the content of all files in the directory then returning an error here makes sense.

Comment (frankquinn) [reply uiServices/src/testCaseServices.go]: Our tech lead asked me to do it this way. She is away at the minute but I will ask her about this again when she returns next week. Thanks for reviewing!

Comment (ConorSmyth) [reply uiServices/src/testCaseServices.go]: @jenny-flynn

**Pull request merged**